tests/components/nibe_heatpump/test_climate.py

```diff
@@ -44,15 +56,249 @@ async def test_basic(
     coils[climate.current] = 20.5
     coils[climate.setpoint_heat] = 21.0
     coils[climate.setpoint_cool] = 30.0
-    coils[climate.mixing_valve_state] = "ON"
+    coils[climate.mixing_valve_state] = 20
     coils[climate.use_room_sensor] = "ON"
-    coils[unit.prio] = "HEAT"
+    coils[unit.prio] = "OFF"
     coils[unit.cooling_with_room_sensor] = "ON"
 
     await async_add_model(hass, model)
 
-    assert hass.states.get(entity_id) == snapshot(name="1. initial")
+    assert hass.states.get(entity_id) == snapshot(name="initial")
+
+    mock_connection.mock_coil_update(unit.prio, "COOLING")
+    assert hass.states.get(entity_id) == snapshot(name="cooling")
+
+    mock_connection.mock_coil_update(unit.prio, "HEAT")
+    assert hass.states.get(entity_id) == snapshot(name="heating")
+
+    mock_connection.mock_coil_update(climate.mixing_valve_state, 30)
+    assert hass.states.get(entity_id) == snapshot(name="idle (mixing valve)")
+
+    mock_connection.mock_coil_update(climate.mixing_valve_state, 20)
+    mock_connection.mock_coil_update(unit.cooling_with_room_sensor, "OFF")
+    assert hass.states.get(entity_id) == snapshot(name="heating (only)")
+
+    mock_connection.mock_coil_update(climate.use_room_sensor, "OFF")
+    assert hass.states.get(entity_id) == snapshot(name="heating (auto)")
+
+    mock_connection.mock_coil_update(unit.prio, None)
+    assert hass.states.get(entity_id) == snapshot(name="off (auto)")
```

Comment:
Can we make a parameterized test out of it?

Reply:
I tried. It get's ugly. It is already parameterized by pump and climate system on the pump. The pumps have different feature sets, and different identifiers for each function. So i can't hardcode identifiers for the climate.mixing_valve_state/cooling..., since their value depend on the pump.

I tried adjusting with some kwargs parameterizization and str -> attr lookup. But it get's unreadable quickly. I also need to mark the snapshots with some understandable identifiers so it's possible to know expected values when you look in the snapshot file.